daras_ai_v2/bot_integration_widgets.py

```diff
@@ -391,14 +393,53 @@ def web_widget_config(bi: BotIntegration, user: AppUser | None):
             if display_pic:
                 bi.photo_url = display_pic
         else:
-            if gui.button(f"{icons.camera} Change Photo"):
-                gui.session_state["--update-display-picture"] = True
-                gui.rerun()
-        bi.name = gui.text_input("###### Name", value=bi.name)
-        bi.descripton = gui.text_area(
-            "###### Description",
-            value=bi.descripton,
+            gencol1, gencol2 = gui.columns(2)
+            with gencol1:
+                if gui.button(f"{icons.camera} Change Photo"):
+                    gui.session_state["--update-display-picture"] = True
+                    gui.rerun()
+            with gencol2:
+                gui.session_state["improve-icon"] = (
+                    '<i class="fa-regular fa-sparkles" aria-hidden="true"></i>'
+                )
+                improve_button = gui.button(
+                    '<i class="fa-regular fa-sparkles" aria-hidden="true"></i> Improve',
+                    "generate_integration_details",
+                    style={"float": "right"},
+                    type="tertiary",
+                )
+
+                if improve_button:
+                    gui.session_state["improve-icon"] = (
+                        '<i class="fa-regular fa-sparkles fa-beat"></i>'
+                    )
+                    result, sr = (
+                        CompareLLMPage()
+                        .get_pr_from_example_id(
+                            example_id=settings.WEB_INTEGRATION_EXAMPLE_ID
+                        )
+                        .submit_api_call(
+                            current_user=user,
+                            request_body=dict(
+                                variables={"title": bi.name, "prompt": copilot_prompt},
+                            ),
+                        )
+                    )
+                    sr.wait_for_celery_result(result)
+                    output_dict = json.loads(sr.state["output_text"]["gpt_4_o"][0])
+                    gui.session_state["bi-description"] = output_dict["description"]
+                    for i in range(4):
+                        gui.session_state[f"--question-{i}"] = output_dict[
+                            "User_question_" + str(i)
```

Comment:
You want to make this schema a list in the json returned. This is a bad way to encode information that you want to index later as an array